catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/DeleteOperations.java

```diff
@@ -260,6 +267,54 @@ private DeleteResponse performLocalDelete(DeleteRequest deleteRequest,
         return deleteResponse;
     }
 
+    //
+    // Private helper methods
+    //
+    private DeleteRequest rewriteQuery(DeleteRequest deleteRequest)
+            throws StopProcessingException, UnsupportedQueryException {
+
+        final String attributeName = deleteRequest.getAttributeName();
+        if (Metacard.ID.equals(attributeName)) {
+            return deleteRequest;
+        }
+
+        Filter filter = frameworkProperties.getFilterBuilder()
+                .anyOf(deleteRequest.getAttributeValues()
+                        .stream()
+                        .map(e -> frameworkProperties.getFilterBuilder()
+                                .attribute(attributeName)
+                                .is()
+                                .equalTo()
+                                .text(e.toString()))
+                        .collect(Collectors.toList()));
+        SourceResponse response = sourceOperations.getCatalog().query(new QueryRequestImpl(new QueryImpl(filter), false));
+        Map<Serializable, Metacard> updateMap = response.getResults()
+                .stream()
+                .map(Result::getMetacard)
+                .filter(MetacardVersion::isNotVersion)
+                .collect(Collectors.toMap(m -> (Serializable) m.getId(), Function.identity()));
+
+        // confirm every item from original update list is present.
+        Set<String> updatedIds = updateMap.values()
+                .stream()
+                .map(m -> m.getAttribute(attributeName))
+                .map(Attribute::getValue)
+                .map(Object::toString)
+                .collect(Collectors.toSet());
+
+        if(!deleteRequest.getAttributeValues()
+                .stream()
+                .allMatch(updatedIds::contains)){
+            LOGGER.warn("Could not obtain metacard ID's for all items in Delete Request!");
+            throw new StopProcessingException("Unable to complete query.");
```

Comment:
Can we provide a better message here?  This will be returned to the endpoint and possibly to a user.  The message in the warning seems close to what we would want to return.